csharp/src/Google.Protobuf/Reflection/FieldDescriptor.cs

```diff
@@ -120,7 +120,7 @@ internal FieldDescriptor(FieldDescriptorProto proto, FileDescriptor file,
             // a MapField, but that feels a tad nasty.
             PropertyName = propertyName;
             Extension = extension;
-            JsonName =  Proto.JsonName == "" ? JsonFormatter.ToJsonName(Proto.Name) : Proto.JsonName;
+            JsonName =  Proto.JsonName.Length == 0 ? JsonFormatter.ToJsonName(Proto.Name) : Proto.JsonName;
```

Comment:
This should be fine (because the value should never be null; it's a string proto field)